src/lib/one-graph/netlify-graph-code-exporter-snippets.js

```diff
@@ -0,0 +1,543 @@
+              const withCoercer = coercerFor(def.type, `event.queryStringParameters?.${name}`)
+              return `const ${munge(name)} = ${withCoercer};`
+            })
+            .join('\n  ') || '',
+        post:
+          dotProp
+            .get(namedOperationData, 'operationDefinition.variableDefinitions', [])
+            .map((def) => {
+              const name = def.variable.name.value
+              return `const ${munge(name)} = eventBodyJson?.${name};`
+            })
+            .join('\n  ') || '',
+      }
+
+      let variableValidation = ''
+
+      let requiredVariableCount = 0
+
+      if ((dotProp.get(namedOperationData, 'operationDefinition.variableDefinitions', []).length !== 0 || 0) > 0) {
+        const requiredVariableNames = namedOperationData.operationDefinition.variableDefinitions
+          .map((def) => (print(def.type).endsWith('!') ? def.variable.name.value : null))
+          .filter(Boolean)
+
+        requiredVariableCount = requiredVariableNames.length
+
+        // TODO: Filter nullable variables
+        const condition = requiredVariableNames
+          .map((name) => `${munge(name)} === undefined || ${munge(name)} === null`)
+          .join(' || ')
+
+        const message = requiredVariableNames.map((name) => `\`${name}\``).join(', ')
+
+        variableValidation = `  if (${condition}) {
+    return {
+      statusCode: 422,
+      body: JSON.stringify({
+        error: 'You must supply parameters for: ${message}'
+      }),
+    };
+  }`
+      }
+
+      return `${pluckerStyle === 'get' ? pluckers.get : pluckers.post}
+
+${requiredVariableCount > 0 ? variableValidation : ''}
+
+  const { errors: ${namedOperationData.name}Errors, data: ${namedOperationData.name}Data } =
+    await Netligraph.${operationFunctionName(namedOperationData)}({ ${invocationParams.join(', ')} }, accessToken);
+
+  if (${namedOperationData.name}Errors) {
+    console.error(JSON.stringify(${namedOperationData.name}Errors, null, 2));
+  }
+
+  console.log(JSON.stringify(${namedOperationData.name}Data, null, 2));`
+    })
+    .join('\n\n')
+
+  return invocations
+}
+
+const clientSideInvocations = (operationDataList, pluckerStyle, useClientAuth) => {
+  const invocations = operationDataList
+    .filter((operationData) => ['query', 'mutation', 'subscription'].includes(operationData.type))
+    .map((namedOperationData) => {
+      const whitespace = 8
+
+      const params = (namedOperationData.operationDefinition.variableDefinitions || []).map(
+        (def) => def.variable.name.value,
+      )
+      let bodyPayload = ''
+
+      if ((dotProp.get(namedOperationData, 'operationDefinition.variableDefinitions', []).length !== 0 || 0) > 0) {
+        const variableNames = namedOperationData.operationDefinition.variableDefinitions.map(
+          (def) => def.variable.name.value,
+        )
+
+        const variables = variableNames.map((name) => `"${name}": ${name}`).join(',\n')
+
+        bodyPayload = `
```

Comment:
If you import `outdent` as `typescript`, you can use the tagged template to get syntax highlighting in some edits, and also it's easier to read.